custom/enikshay/utils.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import absolute_import
+
+from datetime import datetime
+from collections import defaultdict
+
+from django.utils.dateparse import parse_datetime, parse_date
+
+from corehq.apps.fixtures.models import FixtureDataItem, FixtureDataType
+from corehq.form_processor.exceptions import LedgerValueNotFound
+from corehq.form_processor.interfaces.dbaccessors import LedgerAccessors
+from corehq.util.quickcache import quickcache
+from casexml.apps.stock.mock import (
+    Balance,
+    Entry,
+)
+from corehq.apps.hqcase.utils import submit_case_blocks
+from corehq.apps.users.util import SYSTEM_USER_ID
+
+
+def get_adherence_cases_by_date(adherence_cases):
+    adherence_cases_by_date = defaultdict(list)
+    for case in adherence_cases:
+        adherence_date = parse_date(case['adherence_date']) or parse_datetime(case['adherence_date']).date()
```

Comment:
is there the risk that the `case` doesn't have an `adherence_date` property? What happens then?

Reply:
not sure about this. Extracted this method out of the updater. Probably something not expected to happen.
will still add this.